docs/docs/components-and-props.md

```diff
@@ -0,0 +1,248 @@
+
+For example, consider this `Comment` component:
+
+```js
+function Comment(props) {
+  return (
+    <div className="Comment">
+      <div className="UserInfo">
+        <img className="Avatar"
+             src={props.author.avatarUrl}
+             alt={props.author.name} />
+        <div className="UserInfo-name">
+          {props.author.name}
+        </div>
+      </div>
+      <div className="Comment-text">
+        {props.text}
+      </div>
+      <div className="Comment-date">
+        {formatDate(props.date)}
+      </div>
+    </div>
+  );
+}
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/VKQwEo?editors=0010)
+
+It accepts `author` (an object), `text` (a string), and `date` (a date) as props, and describes a comment on a social media website.
+
+This component can be tricky to change because of all the nesting, and it is also hard to reuse individual parts of it. Let's extract a few components from it.
+
+First, we will extract `Avatar`:
+
+```js{3-5}
+function Avatar(props) {
+  return (
+    <img className="Avatar"
+         src={props.user.avatarUrl}
+         alt={props.user.name} />
+  );
+}
+```
+
+The `Avatar` doesn't need to know that it is being rendered inside a `Comment`. This is why we called its prop more generically: `user` rather than `author`. We recommend naming props from the component's own point of view rather than the context in which it is being used.
+
+We can now simplify `Comment` a tiny bit:
+
+```js{5}
+function Comment(props) {
+  return (
+    <div className="Comment">
+      <div className="UserInfo">
+        <Avatar user={props.author} />
+        <div className="UserInfo-name">
+          {props.author.name}
+        </div>
+      </div>
+      <div className="Comment-text">
+        {props.text}
+      </div>
+      <div className="Comment-date">
+        {formatDate(props.date)}
+      </div>
+    </div>
+  );
+}
+```
+
+Next, we will extract a `UserInfo` component that renders an `Avatar` next to user's name:
+
+```js{3-8}
+function UserInfo(props) {
+  return (
+    <div className="UserInfo">
+      <Avatar user={props.user} />
+      <div className="UserInfo-name">
+        {props.user.name}
+      </div>
+    </div>
+  );
+}
+```
+
+This lets us simplify `Comment` even further:
+
+```js{4}
+function Comment(props) {
+  return (
+    <div className="Comment">
+      <UserInfo user={props.author} />
+      <div className="Comment-text">
+        {props.text}
+      </div>
+      <div className="Comment-date">
+        {formatDate(props.date)}
+      </div>
+    </div>
+  );
+}
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/rrJNJY?editors=0010)
+
+Extracting components might seem like grunt work at first, but having a palette of reusable components pays off in larger apps. A good rule of thumb is that if you can come up with a short name (a `Button`, a `Panel`, an `Avatar`, a `FeedStory`) for a part of your app's UI, it is a good candidate to be a reusable component.
```

Comment:
Not sure. I know many people new to react try to write big "screens" because they're used to Backbone templates or similar and don't realize that e.g. a Button is "good enough" to be a component. Maybe need to amend this to say that it some design element comes up more than a few times in different components then it's a good idea.